src/toil_scripts/adam_uberscript/adam_uberscript.py

```diff
@@ -0,0 +1,408 @@
+#!/usr/bin/env python2.7
+"""
+Author: Audrey Musselman-Brown
+Date: 2/12/16
+
+Modified from John Vivian's automated_scaling_tests.py script
+"""
+import logging
+log = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO,
+                    format='%(asctime)-15s:%(levelname)s:%(name)s:%(message)s',
+                    datefmt='%m-%d %H:%M:%S')
+
+import argparse
+from collections import namedtuple
+import csv
+import os
+import random
+import subprocess
+import boto
+from boto.exception import BotoServerError, EC2ResponseError
+import boto.ec2.cloudwatch
+import time
+from uuid import uuid4
+from tqdm import tqdm
+import errno
+from boto_lib import get_instance_ids
+from datetime import datetime, timedelta
+import threading
+import boto.sdb
+from boto.ec2 import connect_to_region
+from datetime import datetime
+from itertools import groupby
+from automated_scaling import *
+
+metric_endtime_margin = timedelta(hours=1)
+metric_initial_wait_period_in_seconds = 0
+metric_collection_interval_in_seconds = 3600
+metric_start_time_margin = 1800
+
+scaling_initial_wait_period_in_seconds = 300
+cluster_scaling_interval_in_seconds = 300
+
+cluster_size_lock = threading.Lock()
+
+aws_region = 'us-west-2'
+
+ref = "https://s3-us-west-2.amazonaws.com/cgl-pipeline-inputs/variant_grch38/GRCh38_full_analysis_set_plus_decoy_hla.fa"
```

Comment:
We're going to have different reference input sets for GRCh38 vs HG19. It might be good to have this in a dictionary instead of in a bunch of globals. This would have the additional benefit that we could swap out the dictionaries easily, IMO.